### DISCUSSION TITLE
Anyone has any idea of how to add untagged/tagged vlans to an interface directly from a switch?

### DISCUSSION THREAD

**Discussion by rfreddy** [netbox-community/netbox#15690] [Help Wanted!]
Greetings, I'm trying to make an ansible playbook that runs this command "show interfaces switchport" that produces an output like this: 

Name: GigabitEthernet 1/1
U       1641
T       1007

I'm quite close capture the data needed for the untagged_vlan and tagged_vlans fields but it seems they don't accept the vlan_id as an input, just the vlan_name. Any suggestion of how to achieve my goal? Any code or suggestion is greatly appreciated.

Thank you and have a great day.

**Comment by candlerb** [comment]
You first have to search for the vlan by vlan_id (tag).

Since Netbox allows you to re-use the same vlan_id, you may get more than one returned. If so, it's up to you to filter the responses or apply additional filters in the query - e.g. by site, by vlan group or whatever allows you to distinguish your broadcast domains.
